components/autofill/src/sync/common.rs

```diff
@@ -222,6 +222,105 @@ macro_rules! sync_merge_field_check {
     };
 }
 
+pub(super) fn common_get_outgoing_records(
+    conn: &Connection,
+    sql: &str,
+    get_payload_from_row: &dyn Fn(&Row<'_>) -> Result<Payload>,
+) -> anyhow::Result<(Vec<Payload>, Vec<(String, String, i64)>)> {
```

Comment:
I was trying to avoid iterating over the rows multiple times but I don't like how complex this is so I'll refactor it.